akd/src/append_only_zks.rs

```diff
@@ -199,88 +206,108 @@ impl Azks {
         Ok(load_count)
     }
 
-    /// An azks is built both by the [crate::directory::Directory] and the auditor.
-    /// However, both constructions have very minor differences, and the append_only_usage
-    /// bool keeps track of this.
-    pub async fn batch_insert_leaves_helper<S: Database + Sync + Send>(
-        &mut self,
+    /// Inserts a batch of leaves recursively from a given node label.
+    #[async_recursion]
+    pub async fn recursive_batch_insert_leaves<S: Database + Sync + Send>(
         storage: &StorageManager<S>,
+        node_label: Option<NodeLabel>,
         insertion_set: Vec<Node>,
+        epoch: u64,
         append_only_exclude_usage: bool,
```

Comment:
nit: I'm going to suggest a @dillonrg suggestion here, can we make this something other than a bool? Like 

```rust

enum AzksInsertUsage {
    Auditor,
    Directory
}
```

Reply:
Addressed in #308, will rebase when it's merged 🙂